Use modern Go constructs, part 2

tklauser opened the pull request:
Follow-up to #38126 and #38157. Change all the remaining instances where modern Go constructs may be used.

Generated using `modernize` by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes with some minor manual edits (e.g. reordering added imports).

The only change that was left out was replacing `omitempty` by `omitzero` in structs since these might introduce API-breaking changes and need to be reviewed individually. This will be done in a separate follow-up PR together with enabling `modernize` in CI.

See individual commit messages for details.

---

Sorry for the huge PR touching many pieces of the code base and requiring many code reviewers :cry: Best reviewed by filtering for "Only files owned by you" so you only see changes for your own review groups:

![image](https://github.com/user-attachments/assets/10a32cde-3eac-459e-9789-8e70e1bd10ad)

tklauser committed:
all: modernize for range loops

Replace any 3-clause for i := 0; i < n; i++ {} loop by for i := range n
{}, added in Go 1.22.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
all: replace interface{} by any

Replace interface{} by the 'any' type added in Go 1.18.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
all: use testing.T.Context

Replace uses of context.WithCancel in tests with testing.T.Context,
added in Go 1.24.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
all: use fmt.Appendf

Replace []byte(fmt.Sprintf(...)) by fmt.Appendf(nil, ...), added in
Go 1.19.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

tklauser committed:
all: use testing.B.Loop for benchmark iteration

Use testing.B.Loop instead of a manual `for range b.N` loop. This also
removes the need to use testing.B.ResetTimer. See
https://github.com/golang/go/issues/61515 for details.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

sayboras reviewed (approved):
Looks good for 6 files, owned my assgined review group.

dylandreimerink reviewed (approved):
Changes look good for my codeowners

smagnani96 reviewed (approved):
All good for the 2 encryption files 👍🏼

bimmlerd commented on pkg/fqdn/cache_test.go:
```diff
@@ -471,12 +469,11 @@ func benchmarkUnmarshalJSON(b *testing.B, numDNSEntries int) {
 	require.NoError(b, err)
 
 	emptyCaches := make([]*DNSCache, b.N)
-	for i := 0; i < b.N; i++ {
```
Not your change, but this is super weird. Can we do something like

```
num := b.N
for i := range num { stuff }
```

instead?

bimmlerd commented on pkg/rate/api_limiter.go:
```diff
@@ -435,14 +435,13 @@ func (l *APILimiter) delayedAdjustment(current, min, max float64) (n float64) {
-		f = 1.0 / l.params.MaxAdjustmentFactor
-	}
-	return f
+	return max(
```
probably correct but I feel like the new code is less readable?

tklauser replied:
Rewritten to

```go
f := l.params.EstimatedProcessingDuration.Seconds() / l.meanProcessingDuration
f = min(f, l.params.MaxAdjustmentFactor)
f = max(f, 1.0/l.params.MaxAdjustmentFactor)
return f
```

which IMO somewhat retains readability while still switching to use `min`/`max`.

bimmlerd reviewed (approved):
two nits, but thanks!

tklauser committed:
all: use min/max built-in functions

Replace if/else conditional assignments by a call to the
built-in min or max functions added in Go 1.21.

Generated using modernize by running:

    go run golang.org/x/tools/gopls/internal/analysis/modernize/cmd/modernize@latest -fix -test ./...

and committing the relevant changes.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

giorio94 reviewed (approved):
Thanks!

nathanjsweet reviewed (approved):
Very modern.
